virttest/utils_disk.py

```diff
@@ -601,6 +603,163 @@ def create_filesystem(session, partition_name, fstype, ostype, timeout=360):
         create_filesyetem_linux(session, partition_name, fstype, timeout)
 
 
+def get_mpoint_fstype_linux(session, partition):
+    """
+    Get a partition's file system type and mountpoint.
+
+    :param session: session object to guest.
+    :param partition: disk partition, like /dev/sdb1.
+    :return: Tuple (mountpoint, fstype)
+    """
+    mount_list = session.cmd_output('cat /proc/mounts')
+    mount_info = re.search(r'%s\s(.+?)\s(.+?)\s' % partition, mount_list)
+    if mount_info:
+        return mount_info.groups()
+    raise exceptions.TestError('partition is not mounted, mounted'
+                               'list as:\n%s' % mount_list)
+
+
+def get_partition_attrs_linux(session, partition):
+    """
+    Get partition attributes in linux guest.
+
+    :param session: session object to guest.
+    :param partition: like /dev/sdb1.
+    :return: dict like {'start': '512B', 'end': '16106127359B',
+                       'size': '16106126848B', 'type': 'primary'}
+    """
+    dev_name, part_num = re.match(r'(/dev/.*)(\d+)', partition).groups()
```

Comment:
Should consider the NVMe type disk.
e.g:
```
[root@ibm-p8-garrison-04 ~]# lsblk
NAME                                MAJ:MIN RM   SIZE RO TYPE MOUNTPOINT
sda                                   8:0    1 931.5G  0 disk 
├─sda1                                8:1    1     4M  0 part 
├─sda2                                8:2    1     1G  0 part /boot
└─sda3                                8:3    1 930.5G  0 part 
  ├─rhel_ibm--p8--garrison--04-root 253:0    0    50G  0 lvm  /
  └─rhel_ibm--p8--garrison--04-home 253:2    0   3.2T  0 lvm  /home
sdb                                   8:16   1 931.5G  0 disk 
└─sdb1                                8:17   1 931.5G  0 part 
  └─rhel_ibm--p8--garrison--04-home 253:2    0   3.2T  0 lvm  /home
sr0                                  11:0    1  1024M  0 rom  
sr1                                  11:1    1  1024M  0 rom  
sr2                                  11:2    1  1024M  0 rom  
sr3                                  11:3    1  1024M  0 rom  
nvme0n1                             259:0    0   1.5T  0 disk 
└─nvme0n1p1                         259:1    0   1.5T  0 part 
  ├─rhel_ibm--p8--garrison--04-swap 253:1    0     4G  0 lvm  [SWAP]
  └─rhel_ibm--p8--garrison--04-home 253:2    0   3.2T  0 lvm  /home

[root@ibm-p8-garrison-04 ~]# parted /dev/nvme0n1
GNU Parted 3.1
Using /dev/nvme0n1
Welcome to GNU Parted! Type 'help' to view a list of commands.
(parted) p                                                                
Model: NVMe Device (nvme)
Disk /dev/nvme0n1: 1600GB
Sector size (logical/physical): 4096B/4096B
Partition Table: msdos
Disk Flags: 

Number  Start   End     Size    Type     File system  Flags
 1      1049kB  1600GB  1600GB  primary               lvm

(parted)           
```

Reply:
updated~ please help review again, thanks~